File: Data Structures & Algorithms/rotate-array/submission-0.go
func rotate(nums []int, k int) {
	k = k % len(nums)
	size := len(nums) - 1
	reverse(nums, 0, size)
	reverse(nums, 0, k - 1)
	reverse(nums, k, size)
}

func reverse(nums []int, left, right int) {
	for left < right {
		nums[left], nums[right] = nums[right], nums[left]
		left++
		right--
	}
}

File: Data Structures & Algorithms/squares-of-a-sorted-array/submission-0.cpp
class Solution {
public:
    vector<int> sortedSquares(vector<int>& nums) {
        int n = nums.size();
        vector<int>res(n);
        int left = 0, right = n - 1, idx = n - 1;
        while ( left <= right ){
            if ( nums[left] * nums[left] < nums[right] * nums[right] ) {
                res[idx] = nums[right] * nums[right];
                right--;
            }else {
                res[idx] = nums[left] * nums[left];
                left++;
            }
            idx--;
        }

        return res;
    }
};